### ISSUE TITLE
[WebToolsE2E][Aspire] After updating .NET SDK to newer band, the Aspire workload does not appear to be installed anymore

### ISSUE THREAD

**Issue by balachir** [dotnet/aspire#2172]
**REPRO STEPS**

1. Clean machine: Win11 x64 23h2 ENU
2. Install latest released .NET SDK 8.0.101
    - From https://dotnet.microsoft.com/en-us/download/dotnet/8.0
3. Install .NET Aspire workload 8.0 Preview 2
    - Open Command Prompt
    - Run `dotnet workload update`
    - Run `dotnet workload install aspire`
    - **Result**: If you run `dotnet --info`, you can see that the Aspire workload is installed properly
    - ![image](https://github.com/dotnet/aspire/assets/8246794/ae8306cd-54b1-4932-8b68-8b6a66e68d23)

4. Install latest released .NET SDK 8.0.200
5. Check status of Aspire workload

**ACTUAL**
If you run `dotnet workload list`, the Aspire workload does not show up anymore

![image](https://github.com/dotnet/aspire/assets/8246794/6387b344-b534-4097-99f2-fd993d898d44)

![image](https://github.com/dotnet/aspire/assets/8246794/d0673d35-a2c0-468a-9136-61c0f137b96a)

**Workaround**:
Re-run `dotnet workload update` and `dotnet workload install aspire` to re-install the aspire workload

**EXPECTED**
The Aspire workload should continue to show up as installed

**Comment by balachir** [comment]
This issue seems to be similar to https://github.com/dotnet/aspire/issues/1559. @nagilson I'm not sure if this will get fixed by the PR https://github.com/dotnet/sdk/pull/38346 that you're working on.

cc: @marcpopMSFT @davidfowl

**Comment by nagilson** [comment]
I dont believe it will be fixed by that, this is a very distinct problem, the other one is due to how workloads managed by VS are handled.

**Comment by marcpopMSFT** [comment]
This is a different issue and a bit of a tougher one. We've talked about whether new bands should recognize workloads installed by prior bands. They are meant to be completely new SDKs (that we don't upgrade customers by default through MU but do for VS) but customers could be frustrated on upgrade. I'm trying to remember if our new first run experience includes --from-previous-sdks @MiYanni 

CC @dsplaisted @baronfel

**Comment by MiYanni** [comment]
@marcpopMSFT 

> I'm trying to remember if our new first run experience includes `--from-previous-sdk`

I didn't know that was a flag until now so no, it doesn't use it. It only runs on an SDK update in the same feature band. Details in the PR description here: https://github.com/dotnet/sdk/pull/34322

**Comment by marcpopMSFT** [comment]
I guess the follow-up question then is should it?

**Comment by v-elenafeng** [comment]
Reproes with the following config:  

- Clean machine  
- Install SDK 8.0.200 
- Install Aspire 8.0 P4 (latest released) 
  - Using CLI, run `dotnet workload update` and `dotnet workload install aspire` 
- Install SDK 8.0.300-preview (latest prerelease build) 
- Check if Aspire workload is installed 
  - Using CLI, run `dotnet workload list`

**Comment by MiYanni** [comment]
@marcpopMSFT 

> I guess the follow-up question then is should it?

I *think* we had this discussion in a triage meeting. If my memory serves me, we decided it didn't make sense to have it update from other feature bands. Additionally, the design for calling this process requires a feature band to be provided as a parameter. So, if work was to be done, either it needs changed or we investigate the logic used for `--from-previous-sdk` to see if we could use that here.

**Comment by marcpopMSFT** [comment]
Thanks for the context. Let's bring up the topic Wednesday to see if we want to include from previous sdks in the logic or leave as is.

**Comment by balachir** [comment]
@joperezr @marcpopMSFT just want to follow-up on this issue. Will this continue to happen once Aspire goes GA? i.e. if users install SDK 8.0.300 + Aspire 8.0 GA + SDK 8.0.400, will this break aspire? If yes, what will be the recommended solution or workaround?

**Comment by balachir** [comment]
Thanks @marcpopMSFT. I gave this a try using the latest build of SDK 8.0.300. But I can't get it to work as I expected. With workload sets, I thought that the `dotnet workload update --from-previous-sdk` command in step 10 below will automatically restore the aspire workload for SDK 8.0.300. Did I not understand it correctly?

cc: @joperezr 

Steps
1. Clean Win11 machine 
1. Install SDK 8.0.204
1. Open Command Prompt 
1. Run `dotnet workload update`
1. Run `dotnet workload install aspire` - I get Aspire version 8.0.0-preview.6.24214.1
1. Run `dotnet workload list` -- works fine
1. Run `dotnet new aspire-starter -o AspireStarterApp01` -- works fine
1. Install **SDK 8.0.300**
1. Run `dotnet workload list` -- does not list Aspire workload. This is expected
1. Run `dotnet workload update --from-previous-sdk` -- command completes fine but there's still no aspire workload
![image](https://github.com/dotnet/aspire/assets/8246794/a76cb500-2bef-4396-9dde-997e2e7c7e9a)
1. Run `dotnet workload install aspire` -- now I can see that the Aspire P6 workload is available again
![image](https://github.com/dotnet/aspire/assets/8246794/e75758b9-55d0-4ee8-b76c-45042aca0434)

**Comment by marcpopMSFT** [comment]
@dsplaisted please take a look as I would expect from-previous-sdk to work for this scenario. I'm wondering if it's some nuance of the previews causing the issue.

**Comment by v-elenafeng** [comment]
@marcpopMSFT @balachir This issue reproes again with SDK 8.0.303 + Aspire 8.1.0 + SDK 8.0.400

3 points on its severity
- This is a 'dotnet workload list' only issue, with the issue reproducing, we can still create the aspire projects.
- The workaround is re-run 'dotnet workload update' and 'dotnet workload install aspire' to re-install the aspire workload.
- Doesn't repro with SDK update from 8.0.302 to 8.0.303/8.0.304, so I assume it should not be able to repro with SDK 8.0.400 updating to 8.0.401 either.

**Comment by balachir** [comment]
@v-elenafeng, for the repro steps that you described, I believe this is currently by design. See Marc's earlier comments [here](https://github.com/dotnet/aspire/issues/2172#issuecomment-2065219625). If you have questions, we can chat more during our next sync meeting.

**Comment by joperezr** [comment]
Sorry for not closing this issue sooner, as Marc called out above, this is by design. It is also not an issue any longer for Aspire customers given Aspire no longer uses an SDK workload starting from 9.0. Closing this as by design.
